File: backend/OrderService/src/main/java/com/clickmunch/OrderService/entity/Order.java
package com.clickmunch.OrderService.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("orders")
public class Order {
    @Id
    private Long id;
    private Long customerId;
    private String customerName;
    private Long restaurantId;
    private String restaurantName;
    private OrderStatus status;
    private OrderChannel channel;
    private String notes;
    private String eta;
    private BigDecimal total;
    private Long tableId;
    private Long waiterId;
    private BigDecimal tipAmount;
    private String waiterComment;
    private Integer preparationMinutes;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/mobile/core/orders/interface/order.ts
export type OrderStatus =
  | 'Pending'
  | 'SentToKitchen'
  | 'Preparing'
  | 'Ready'
  | 'Served'
  | 'Delivered'
  | 'Cancelled';

export type OrderChannel = 'Reservation' | 'In-person';

export interface OrderItem {
  id?: number;
  orderId?: number;
  menuItemId: string;
  productName: string;
  quantity: number;
  unitPrice: number;
  subtotal: number;
}

export interface Order {
  id: number;
  customerId: number;
  customerName: string;
  restaurantId: number;
  restaurantName: string;
  status: OrderStatus;
  channel: OrderChannel;
  notes?: string;
  eta?: string;
  total: number;
  tableId?: number;
  waiterId?: number;
  tipAmount?: number;
  waiterComment?: string;
  preparationMinutes?: number;
  items: OrderItem[];
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateOrderRequest {
  customerId: number;
  customerName: string;
  restaurantId: number;
  restaurantName: string;
  channel: OrderChannel;
  notes?: string;
  items: {
    menuItemId: string;
    productName: string;
    quantity: number;
    unitPrice: number;
  }[];
}
